### DISCUSSION TITLE
Why does vomnibar's theme randomly switch to dark even after pressing `gn`?

### DISCUSSION THREAD

**Discussion by txtyash** [gdh1995/vimium-c#706] [Q&A] [answered]
Even after pressing `gn`, vomnibar frequently switches to dark mode. Not sure why. Is it because the browser is using a dark theme?

**Comment by gdh1995** [comment]
Um, yes it's because `gn` was implemented before browsers support dark mode, so it doesn't consider about the dark theme when computing its priority.

**Reply by 2tolik** [reply]
gn - is a workaround for "system theme" glitch that only affects Vomnibar (and ?) on my MacOS (works ok on Win11)

edge://settings/appearance > system 
"Dark reader" extension "use system" 
Vimium-C - auto-switch is fully on

result: when system theme toggled on macos - everything switches, even Options page for Vimium-C, except Vomnibar style :(
workaround: gn or mouse (hover over vomnibar, right side) fixes it

**Reply by gdh1995** [reply]
Vimium C can not respond on dark mode changes immediately - the Options page succeeds just because it's a normal web page, while the main part of Vimium C is not.

Normally, Vimium C's UI (LinkHints marks, HUD and Vominbar) should turn into black/white in up to 1 minute
- yes Vimium C uses an ugly timer to query the theme state every 60 seconds

But, if you have triggered `gn`, then Vomnibar theme may be locked and then never respond on system theme changes until your browser restarting.

Apparently the logic is too confusing, so I'll rewrite it in the future (2024 or 2025...).

**Comment by gdh1995** [comment, marked as answer]
I'm not sure how to fix its logic. If you usually use dark theme, then you may add "`dark`" to `Vomnibar settings`:
![image](https://user-images.githubusercontent.com/5547703/185096226-6973e5d8-9395-4b27-b07a-aa1b2fe085fd.png)

And then `gn` should be able to override the effect of dark theme.
